assets/chat/js/emotes.js

```diff
@@ -1,42 +1,59 @@
 export default class EmoteService {
-  emotes = [];
+  emotes = new Map();
 
   regexForEmotes(emotes) {
-    const prefixes = emotes.map((e) => e.prefix);
+    const prefixes = [...emotes].map((e) => e.prefix);
     return new RegExp(`(^|\\s)(${prefixes.join('|')})(?=$|\\s)`, 'gm');
   }
 
   emoteRegexForUser(user) {
-    if (user.isPrivileged()) return this.regexForEmotes(this.emotes);
-    let emotes = this.emotes.filter(
+    if (user.isPrivileged()) return this.regexForEmotes(this.emotes.values());
+
+    let emotes = [...this.emotes.values()].filter(
       (e) => e.minimumSubTier <= user.subTier && !e.twitch
     );
-    if (user.isTwitchSub())
-      emotes = emotes.concat(this.emotes.filter((e) => e.twitch));
+
+    if (user.isTwitchSub()) {
+      emotes = emotes.concat([...this.emotes.values()].filter((e) => e.twitch));
+    }
+
     return this.regexForEmotes(emotes);
   }
 
   get prefixes() {
-    return this.emotes.map((e) => e.prefix);
+    return [...this.emotes.keys()];
   }
 
   get systemEmoteRegex() {
-    return this.regexForEmotes(this.emotes);
+    return this.regexForEmotes(this.emotes.values());
   }
 
   get twitchEmotePrefixes() {
     return this.emotes.filter((e) => e.twitch).map((e) => e.prefix);
   }
 
   getEmote(emote) {
-    return this.emotes.find(
-      (e) => e.prefix.toLowerCase() === emote.toLowerCase()
-    );
+    if (this.emotes.has(emote)) return this.emotes.get(emote);
+    return null;
```

Comment:
I'm fairly certain `Map.get()` simply returns `undefined` if the key doesn't exist. If so, you don't need this extra check to see if it exists.
```suggestion
    return this.emotes.get(emote);
```